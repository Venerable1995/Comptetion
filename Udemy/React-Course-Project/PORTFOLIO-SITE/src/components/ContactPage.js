import React from "react";


const ContactPage = ()=>(
    <div>
    <p> this is my contact page</p>
    </div>
)
export default ContactPage;