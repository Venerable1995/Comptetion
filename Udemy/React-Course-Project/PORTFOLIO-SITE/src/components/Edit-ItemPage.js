import React from "react";


const EditItemPage= (props)=>(
    <div>
    <p>this is my Portfolio of item : {(props.match.params.id)} </p>
    
</div>
)
export default EditItemPage;