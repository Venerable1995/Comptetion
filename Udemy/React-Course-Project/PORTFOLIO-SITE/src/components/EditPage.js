import React from "react";
import {Link} from 'react-router-dom';
const EditPage= ()=>(
    <div>
    <h1> My Work  </h1>
    <p> checkout the stuff i've done </p>
    <Link to="/portfolio/:1">item-one</Link>
</div>
)
export default EditPage;