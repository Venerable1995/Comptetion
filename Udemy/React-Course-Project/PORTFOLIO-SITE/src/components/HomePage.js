import React from "react";

const HomePage= ()=>(
    <div>
    <p>this is my home page</p>
    </div>
)

export default HomePage;