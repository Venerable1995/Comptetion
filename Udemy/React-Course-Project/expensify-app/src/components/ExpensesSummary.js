import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary= (props)=>(
    <div>
   { props.expenses.length === 0 ? (<div/>) 
    :
    <h1>  
      Viewing {props.expenses.length} Expense Totalling : { props.expenses.map((expense)=>expense.amount).reduce(( sum , value )=> sum + value /100, 0)}     
    </h1>
    }
    </div>
)

const mapStateToProps = (state) => {
    return {
      expenses: selectExpenses(state.expenses, state.filters)
    };
  };
  
  export default connect(mapStateToProps)(ExpensesSummary);
  