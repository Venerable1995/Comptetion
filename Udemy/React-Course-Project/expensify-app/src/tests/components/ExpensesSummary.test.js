import React from 'react';
import { shallow } from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should view totalling of 1 item',()=>{
const res = shallow( <ExpensesSummary expenses={expenses} />)
expect(res).toMatchSnapshot();
})