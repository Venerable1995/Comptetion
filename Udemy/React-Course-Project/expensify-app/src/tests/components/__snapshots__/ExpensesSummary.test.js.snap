// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should view totalling of 1 item 1`] = `
<div>
  <h1>
    Viewing 
    3
     Expense Totalling : 
    1141.95
  </h1>
</div>
`;
